docs/setup/config.md

```diff
@@ -272,7 +272,9 @@ For example when running Flink on YARN on an environment with a restrictive fire
 
 - `recovery.zookeeper.quorum`: Defines the ZooKeeper quorum URL which is used to connet to the ZooKeeper cluster when the 'zookeeper' recovery mode is selected
 
-- `recovery.zookeeper.path.root`: (Default '/flink') Defines the root dir under which the ZooKeeper recovery mode will create znodes.
+- `recovery.zookeeper.path.root`: (Default '/flink') Defines the root dir under which the ZooKeeper recovery mode will create namespace directories.
+
+- `recovery.zookeeper.path.namespace`: (Default '/default_ns' in standalone mode, or the <yarn-application-id> under Yarn) Defines the subdirectory under the root dir where the ZooKeeper recovery mode will create znodes. This allows to isolate multiple applications on the same ZooKeeper. 
```

Comment:
You're right, it would just be a heuristic for generating a unique namespace but it is not unique because multiple Flink instances might be running on the same host names (e.g. multiple processes, Docker containers). So +1 for not adding something that can cause hard to debug problems. We have covered the Yarn use case which makes it very convenient to setup a namespace.